src/prng/os-prng.lisp

```diff
@@ -0,0 +1,25 @@
+;;;; -*- mode: lisp; indent-tabs-mode: nil -*-
+;;;; os-prng.lisp -- OS-provided pseudo-random number generator
+
+(in-package :crypto)
+
+(defclass os-prng ()
+  (#+unix source))
+
+(defmethod prng-random-data (num-bytes (prng os-prng))
+  #+unix
+  (let ((seq (make-array num-bytes :element-type 'unsigned-byte)))
+    (unless (slot-boundp prng 'source)
+      (setf (slot-value prng 'source)
+            (open #P"/dev/urandom" :element-type 'unsigned-byte)))
+    (assert (>= (read-sequence seq (slot-value prng 'source)) num-bytes))
+    seq)
+  ;; FIXME: this is _untested_!
```

Comment:
One major issue here is that only SBCL is supported on Windows. I have obtained a Windows computer and will attempt to figure out how to call CryptGenRandom from other implementations.

If support for other implementations on Windows is required, this PR needs to remain open.